EIPS/eip-4938.md

```diff
@@ -0,0 +1,57 @@
+last-call-deadline: 
+type: Standards Track
+category: Networking
+created: 2022-03-23
+requires: 2464, 2482
+---
+
+## Abstract
+
+The [Ethereum Wire Protocol][eth] defines request and response messages for exchanging data between clients. The `GetNodeData` request retrieves a set of trie nodes or contract code from the state trie by hash. We propose to remove the `GetNodeData` and `NodeData` messages from the wire protocol.
+
+## Motivation
+
+`GetNodeData` and `NodeData` were introduced in protocol version `eth/63` to allow for a sync mode called "fast sync", which downloads the Ethereum state without executing all blocks. The sync algorithm works by requesting all state trie nodes and contract codes by their hash.
+
+Serving `GetNodeData` requests requires clients to store state as a mapping of hashes to trie nodes. Avoiding the need to store such a mapping in the database is the main motivation for removing this request type.
+
+Some client implementations, notably OpenEthereum and Erigon, do not serve `GetNodeData` at this time, because they do not store the state in a compatible way. The Ethereum Wire Protocol should accurately reflect the capabilites of clients, and should not contain messages which are impossible to implement in some clients.
+
+## Specification
+
+Remove the following message types from the `eth` protocol:
+
+* `GetNodeData (0x0d)`
+   * **(eth/66)**: `[request_id: P, [hash_0: B_32, hash_1: B_32, ...]]`
+* `NodeData (0x0e)`
+   * **(eth/66)**: `[request_id: P, [value_0: B, value_1: B, ...]]`
+
+## Rationale
+
+The Geth team has discussed this proposal with the developers of Erigon, Nethermind and Besu. We agreed `GetNodeData` is a roadblock that needs to be removed in order to improve state database implementations.
+
+A replacement for `GetNodeData` is available in the [snap protocol][snap]. Specifically, clients can use the [GetByteCodes][snap-getbytecodes] and [GetTrieNodes][snap-gettrienodes] messages instead of `GetNodeData`. The snap protocol can be used to implement the "fast sync" algorithm, though it is recommended to use it for "snap sync".
+
+## Backwards Compatibility
+
+This EIP changes the `eth` protocol and requires rolling out a new version, `eth/67`. Supporting multiple versions of a wire protocol is possible. Rolling out a new version does not break older clients immediately, since they can keep using protocol version `eth/66`.
+
+This EIP does not change consensus rules or the EVM and does not require a hard fork.
+
+## Security Considerations
+
+None
+
+## Copyright
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+[eth]: https://github.com/ethereum/devp2p/tree/master/caps/eth.md
+[snap]: https://github.com/ethereum/devp2p/tree/master/caps/snap.md
+[snap-getbytecodes]: https://github.com/ethereum/devp2p/blob/master/caps/snap.md#getbytecodes-0x04
+[snap-gettrienodes]: https://github.com/ethereum/devp2p/blob/master/caps/snap.md#gettrienodes-0x06
```

Comment:
Please inline links, rather than using a references section.  This is especially true for links that only have one instance in the document (like all of these).